include/libtorrent/bencode.hpp

```diff
@@ -88,6 +88,28 @@ namespace libtorrent
 
 	namespace detail
 	{
+		template <typename T, typename Cond = typename std::enable_if<
```

Comment:
is the main purpose here to support arbitrary iterators, and not just char pointers?